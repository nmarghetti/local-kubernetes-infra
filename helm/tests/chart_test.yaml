templates:
  - deployments.yaml
  - services.yaml
  - ingress.yaml
tests:
  - it: default chart deployment should have no document
    asserts:
      - hasDocuments:
          count: 0
        template: deployments.yaml
  - it: the deployment should have 2 deployments and 1 service
    set:
      apps.backend.deployment.enabled: true
      apps.backend.service.enabled: true
      apps.frontend.deployment.enabled: true
    values:
      - values-base.yaml
    asserts:
      - hasDocuments:
          count: 2
        template: deployments.yaml
      - hasDocuments:
          count: 1
        template: services.yaml
  - it: chart should match snapshot
    release:
      name: mychart
    chart:
      version: 0.0.1
    values:
      - values-base.yaml
      - values-enabled.yaml
    asserts:
      - matchSnapshot: {}
