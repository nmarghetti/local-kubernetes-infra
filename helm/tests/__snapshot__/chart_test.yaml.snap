chart should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        kubectl.kubernetes.io/default-container: app
      labels:
        app.kubernetes.io/instance: mychart
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: play
        app.kubernetes.io/version: na
        helm.sh/chart: mychart-0.0.1
      name: mychart-play-backend
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/component: backend
          app.kubernetes.io/instance: mychart
          app.kubernetes.io/name: play
      template:
        metadata:
          labels:
            app.kubernetes.io/component: backend
            app.kubernetes.io/instance: mychart
            app.kubernetes.io/name: play
        spec:
          containers:
            - image: stefanprodan/podinfo:latest
              imagePullPolicy: IfNotPresent
              name: app
              ports:
                - containerPort: 8080
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        kubectl.kubernetes.io/default-container: app
      labels:
        app.kubernetes.io/instance: mychart
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: play
        app.kubernetes.io/version: na
        helm.sh/chart: mychart-0.0.1
      name: mychart-play-frontend
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/component: frontend
          app.kubernetes.io/instance: mychart
          app.kubernetes.io/name: play
      template:
        metadata:
          labels:
            app.kubernetes.io/component: frontend
            app.kubernetes.io/instance: mychart
            app.kubernetes.io/name: play
        spec:
          containers:
            - image: docker.io/frontend:latest
              imagePullPolicy: IfNotPresent
              name: app
              ports:
                - containerPort: 8080
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/component: backend
        app.kubernetes.io/instance: mychart
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: play
        app.kubernetes.io/version: na
        helm.sh/chart: mychart-0.0.1
      name: mychart-play-backend
    spec:
      ports:
        - port: 8080
          targetPort: 8080
      selector:
        app.kubernetes.io/component: backend
        app.kubernetes.io/instance: mychart
        app.kubernetes.io/name: play
      type: ClusterIP
  4: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/component: frontend
        app.kubernetes.io/instance: mychart
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: play
        app.kubernetes.io/version: na
        helm.sh/chart: mychart-0.0.1
      name: mychart-play-frontend
    spec:
      selector:
        app.kubernetes.io/component: frontend
        app.kubernetes.io/instance: mychart
        app.kubernetes.io/name: play
      type: ClusterIP
  5: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: mychart
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: play
        app.kubernetes.io/version: na
        helm.sh/chart: mychart-0.0.1
      name: mychart-play-backend
    spec:
      rules:
        - host: play.local
          http:
            paths:
              - backend:
                  service:
                    name: mychart-play-backend
                    port:
                      number: 8080
                path: /
                pathType: Prefix
  6: |
    apiVersion: traefik.io/v1alpha1
    kind: IngressRoute
    metadata:
      annotations:
        kubernetes.io/ingress.class: traefik
      labels:
        app.kubernetes.io/instance: mychart
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: play
        app.kubernetes.io/version: na
        helm.sh/chart: mychart-0.0.1
      name: mychart-play-frontend
    spec:
      routes:
        - kind: Rule
          match: Host(`play.local`)
          services:
            - kind: Service
              name: mychart-play-frontend
              port: 8080
